rtt/Activity.cpp

```diff
@@ -189,6 +191,11 @@ namespace RTT
                 mtimeout = false;
                 this->step();
                 this->work(base::RunnableInterface::TimeOut);
+                // get the current time before sleeping again
+                if (wakeup != 0 &&
+                        this->getTask()->wait_policy == ORO_WAIT_ABS) {
```

Comment:
It is not allowed to use internal fields of the `RTOS_TASK` struct outside the respective OS abstraction.